### PR TITLE
chore(ci): added docker security scan and a linter

### PR DESCRIPTION
**Pull request by hutchic**
added hadolint with a few exceptions
added the docker snyk security scan feature with exceptions
the community is beginning to sway towards it being acceptable to do upgrades in images

- https://github.com/hadolint/hadolint/pull/677
- https://github.com/hadolint/hadolint/issues/562
- https://github.com/OWASP/CheatSheetSeries/pull/614

### CODE REVIEW

File: alpine/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.13
+FROM alpine:3.14
```
Comment (hutchic) [inline alpine/Dockerfile:1]:
```suggestion
FROM alpine:3.13
```

File: tests/01-image.test.sh [hunk 2]
```diff
@@ -3,7 +3,12 @@
 function run_test {
   tinitialize "Docker-Kong test suite" "${BASH_SOURCE[0]}"
 
+  docker run -i --rm -v $PWD/hadolint.yaml:/.config/hadolint.yaml hadolint/hadolint < $BASE/Dockerfile
```
Comment (gszr) [inline tests/01-image.test.sh:6]: should we pin?

File: tests/01-image.test.sh [hunk 3]
```diff
@@ -3,7 +3,12 @@
...
 
+  if [[ ! -z "${SNYK_SCAN_TOKEN}" ]]; then
```
Comment (gszr) [inline tests/01-image.test.sh:8]: checking: is the token added to CI?

**Review by gszr — APPROVED** [review]
great one!

File: rhel/Dockerfile [hunk 4]
```diff
@@ -1,9 +1,7 @@
-ARG RHEL_VERSION=8
+ARG RHEL_VERSION=7
```
Comment (Tieske) [inline rhel/Dockerfile]: is the downgrade intended?

Comment (hutchic) [reply rhel/Dockerfile]: yeah I upgraded as part of the PR and that broke tests. There were no security issues in 7 so going to keep us pinned there for now

File: tests/01-image.test.sh (hunk 3 above)
Comment (hutchic) [reply tests/01-image.test.sh:8]: It is

File: tests/01-image.test.sh (hunk 2 above)
Comment (hutchic) [inline tests/01-image.test.sh:6]:
```suggestion
  docker run -i --rm -v $PWD/hadolint.yaml:/.config/hadolint.yaml hadolint/hadolint:2.7.0 < $BASE/Dockerfile
```

Comment (hutchic) [reply tests/01-image.test.sh:6]: pin the hadolint scanner

**Pull request merged**